mousestyles/GLRT_distribution.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import print_function, absolute_import, division
+
+import numpy as np
+from mousestyles import data
+
+
+def powerlaw_pdf(x, a):
+    """
+    The probability density function of truncated power law.
+
+    Parameters
+    ----------
+    x : int
+        x in formula p(x) = (alpha-1)*x^(-alpha).
+    a : int>1
+        alpha in formula p(x) = (alpha-1)*x^(-alpha).
+
+    Returns
+    -------
+    probability density : a float number
+        The probability density of power law at x.
+
+    Examples
+    --------
+    >>> powerlaw_pdf (2,2)
+    0.25
+    """
+    return((a - 1) * x**(-a))
+
+
+def exp_pdf(x, l):
+    """
+    The probability density function of truncated exponential.
+
+    Parameters
+    ----------
+    x : int
+        x in formula p(x) = lambda*exp(-lambda*x).
+    l : int
+        lambda in formula p(x) = lambda*exp(-lambda*x).
+
+    Returns
+    -------
+    probability density : a float number
+        The probability density of power law at x.
+
+    Examples
+    --------
+    >>> exp_pdf
+    0.36787944117144233
+    """
+    return(l * np.exp(-l * (x - 1)))
+
+
+def random_powerlaw(n, a):
+    """
+    Random generate points of truncated power law.
+
+    Parameters
+    ----------
+    n : int
+        number of points
+    a : int>1
+        power law parameter alpha
+
+    Returns
+    -------
+    points : a vector of float number
+        n points have the target distribution.
+
+    Examples
+    --------
+    >>> random_powerlaw(4,2)
+    array([  1.18097435,   1.04584078,   1.4650779 ,  36.03967524])
+    """
+    y = np.random.sample(n)
+    return((1 - y)**(-1 / (a - 1)))
+
+
+def random_exp(n, l):
+    """
+    Random generate points of truncated exponential.
+
+    Parameters
+    ----------
+    n : int
+        number of points
+    l : int
+        exponential parameter lambda
+
+    Returns
+    -------
+    points : a vector of float number
+        n points have the target distribution.
+
+    Examples
+    --------
+    >>> random_exp(4,2)
+    array([ 1.07592496,  1.19789646,  1.19759663,  1.03993227])
+    """
+    y = np.random.exponential(1.0 / l, n)
+    return(y + 1)
+
+
```

Comment:
Thank you for your advice. We can definitely change the function, however, it will be quite lengthy as these two function has different test statistics, two simulation methods to find p-value and also two different input options. In brief, there is nothing similar between these two function and combining two function does not make it any faster or save space; instead, it will make each sentence too long and hard to read. Thus I choose not to write it as one function.